oak-pojosr has a test failure with:
```
Caused by: org.h2.jdbc.JdbcSQLDataException: Value too long for column "BINARY VARYING": "3b4e0d35d7bda3579118fed83f2c7ba06fbd1c606a6dede074c9b2a980fa84d78613cdc1774af692... (2097152)" [22001-206]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:525)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:496)
	at org.h2.message.DbException.get(DbException.java:227)
	at org.h2.message.DbException.getValueTooLongException(DbException.java:341)
	at org.h2.value.ValueVarbinary.<init>(ValueVarbinary.java:34)
	at org.h2.value.ValueVarbinary.getNoCopy(ValueVarbinary.java:65)
	at org.h2.value.ValueVarbinary.get(ValueVarbinary.java:51)
	at org.h2.jdbc.JdbcPreparedStatement.setBytes(JdbcPreparedStatement.java:1001)
	at org.apache.jackrabbit.oak.plugins.document.rdb.RDBBlobStore.storeBlockInDatabase(RDBBlobStore.java:334)
```
Looks like 1MB is the limit for BINARY VARYING: https://h2database.com/html/advanced.html#limits_limitations

--- a/oak-pojosr/src/test/groovy/org/apache/jackrabbit/oak/run/osgi/DocumentNodeStoreConfigTest.groovy
+++ b/oak-pojosr/src/test/groovy/org/apache/jackrabbit/oak/run/osgi/DocumentNodeStoreConfigTest.groovy
@@ -28,6 +28,7 @@ import org.apache.jackrabbit.oak.plugins.document.DocumentStoreStatsMBean
 import org.apache.jackrabbit.oak.plugins.document.MongoUtils
 import org.apache.jackrabbit.oak.plugins.document.mongo.MongoBlobStore
 import org.apache.jackrabbit.oak.plugins.document.util.MongoConnection
+import org.apache.jackrabbit.oak.spi.blob.AbstractBlobStore
 import org.apache.jackrabbit.oak.spi.blob.BlobStore
 import org.apache.jackrabbit.oak.spi.blob.GarbageCollectableBlobStore
 import org.apache.jackrabbit.oak.spi.blob.MemoryBlobStore
@@ -70,6 +71,9 @@ class DocumentNodeStoreConfigTest extends AbstractRepositoryFactoryTest {
         ])
 
         DocumentNodeStore ns = getServiceWithWait(NodeStore.class)
+        // OAK-9668: H2 2.0.206 has a limit of 1MB for BINARY VARYING
+        AbstractBlobStore blobStore = getServiceWithWait(BlobStore.class)
+        blobStore.setBlockSize(1024 * 1024)
 
         //3. Check that DS contains tables from both RDBBlobStore and RDBDocumentStore
         assert getExistingTables(ds).containsAll(['NODES', 'DATASTORE_META'])

--- a/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/persistentCache/DataTypeUtil.java
+++ b/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/persistentCache/DataTypeUtil.java
@@ -32,6 +32,7 @@ import org.apache.jackrabbit.oak.plugins.document.Revision;
 import org.apache.jackrabbit.oak.plugins.document.RevisionVector;
 import org.h2.mvstore.DataUtils;
 import org.h2.mvstore.WriteBuffer;
+import org.h2.mvstore.type.DataType;
 import org.h2.mvstore.type.StringDataType;
 
 /**
@@ -166,4 +167,40 @@ class DataTypeUtil {
         return new DocumentNodeState(store, p, rootRevision, props,
                 !noChildren, mem, lastRevision, false);
     }
+
+    /**
+     * Cast the storage object to an array of type T.
+     *
+     * @param storage the storage object
+     * @return the array
+     */
+    static Object[] cast(Object storage) {
+        return (Object[])storage;
+    }
+
+    static int binarySearch(DataType<Object> dataType, Object key, Object storageObj, int size, int initialGuess) {
+        Object[] storage = cast(storageObj);
+        int low = 0;
+        int high = size - 1;
+        // the cached index minus one, so that
+        // for the first time (when cachedCompare is 0),
+        // the default value is used
+        int x = initialGuess - 1;
+        if (x < 0 || x > high) {
+            x = high >>> 1;
+        }
+        while (low <= high) {
+            int compare = dataType.compare(key, storage[x]);
+            if (compare > 0) {
+                low = x + 1;
+            } else if (compare < 0) {
+                high = x - 1;
+            } else {
+                return x;
+            }
+            x = (low + high) >>> 1;
+        }
+        return ~low;
+    }
+
 }

--- a/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/persistentCache/KeyDataType.java
+++ b/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/persistentCache/KeyDataType.java
@@ -16,6 +16,8 @@
  */
 package org.apache.jackrabbit.oak.plugins.document.persistentCache;
 
+import static org.apache.jackrabbit.oak.plugins.document.persistentCache.DataTypeUtil.cast;
+
 import java.nio.ByteBuffer;
 
 import org.apache.jackrabbit.oak.cache.CacheValue;
@@ -64,40 +66,9 @@ public class KeyDataType implements DataType<Object> {
         }
     }
 
-    /**
-     * Cast the storage object to an array of type T.
-     *
-     * @param storage the storage object
-     * @return the array
-     */
-    protected final Object[] cast(Object storage) {
-        return (Object[])storage;
-    }
-
     @Override
-    public int binarySearch(Object key, Object storageObj, int size, int initialGuess) {
-        Object[] storage = cast(storageObj);
-        int low = 0;
-        int high = size - 1;
-        // the cached index minus one, so that
-        // for the first time (when cachedCompare is 0),
-        // the default value is used
-        int x = initialGuess - 1;
-        if (x < 0 || x > high) {
-            x = high >>> 1;
-        }
-        while (low <= high) {
-            int compare = compare(key, storage[x]);
-            if (compare > 0) {
-                low = x + 1;
-            } else if (compare < 0) {
-                high = x - 1;
-            } else {
-                return x;
-            }
-            x = (low + high) >>> 1;
-        }
-        return ~low;
+    public int binarySearch(Object key, Object storage, int size, int initialGuess) {
+        return DataTypeUtil.binarySearch(this, key, storage, size, initialGuess);
     }
 
     @Override

--- a/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/persistentCache/ValueDataType.java
+++ b/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/persistentCache/ValueDataType.java
@@ -16,6 +16,8 @@
  */
 package org.apache.jackrabbit.oak.plugins.document.persistentCache;
 
+import static org.apache.jackrabbit.oak.plugins.document.persistentCache.DataTypeUtil.cast;
+
 import java.nio.ByteBuffer;
 
 import org.apache.jackrabbit.oak.cache.CacheValue;
@@ -72,40 +74,9 @@ public class ValueDataType implements DataType<Object> {
         }
     }
     
-    /**
-     * Cast the storage object to an array of type T.
-     *
-     * @param storage the storage object
-     * @return the array
-     */
-    protected final Object[] cast(Object storage) {
-        return (Object[])storage;
-    }
-
     @Override
-    public int binarySearch(Object key, Object storageObj, int size, int initialGuess) {
-        Object[] storage = cast(storageObj);
-        int low = 0;
-        int high = size - 1;
-        // the cached index minus one, so that
-        // for the first time (when cachedCompare is 0),
-        // the default value is used
-        int x = initialGuess - 1;
-        if (x < 0 || x > high) {
-            x = high >>> 1;
-        }
-        while (low <= high) {
-            int compare = compare(key, storage[x]);
-            if (compare > 0) {
-                low = x + 1;
-            } else if (compare < 0) {
-                high = x - 1;
-            } else {
-                return x;
-            }
-            x = (low + high) >>> 1;
-        }
-        return ~low;
+    public int binarySearch(Object key, Object storage, int size, int initialGuess) {
+        return DataTypeUtil.binarySearch(this, key, storage, size, initialGuess);
     }
 
     @Override

--- a/oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/blob/RDBBlobStoreTest.java
+++ b/oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/blob/RDBBlobStoreTest.java
@@ -19,6 +19,7 @@ package org.apache.jackrabbit.oak.plugins.document.blob;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeTrue;
 
 import java.io.IOException;
 import java.security.MessageDigest;
@@ -76,6 +77,7 @@ public class RDBBlobStoreTest extends AbstractBlobStoreTest {
         return result;
     }
 
+    private RDBBlobStoreFixture fixture;
     private RDBBlobStore blobStore;
     private String blobStoreName;
     private RDBDataSourceWrapper dsw;
@@ -83,6 +85,7 @@ public class RDBBlobStoreTest extends AbstractBlobStoreTest {
     private static final Logger LOG = LoggerFactory.getLogger(RDBBlobStoreTest.class);
 
     public RDBBlobStoreTest(RDBBlobStoreFixture bsf) {
+        fixture = bsf;
         blobStore = bsf.createRDBBlobStore();
         blobStoreName = bsf.getName();
         dsw = bsf.getDataSource();
@@ -118,6 +121,9 @@ public class RDBBlobStoreTest extends AbstractBlobStoreTest {
 
     @Test
     public void testBigBlob() throws Exception {
+        // OAK-9668: H2 has a limit of 1MB for BINARY VARYING
+        assumeTrue(fixture != RDBBlobStoreFixture.RDB_H2);
+
         int min = 0;
         int max = 8 * 1024 * 1024;
         int test = 0;
@@ -146,7 +152,7 @@ public class RDBBlobStoreTest extends AbstractBlobStoreTest {
 
         LOG.info("max blob length for " + blobStoreName + " was " + test);
 
-        int expected = Math.max(blobStore.getBlockSize(), 1 * 1024 * 1024);
+        int expected = Math.max(blobStore.getBlockSize(), 2 * 1024 * 1024);
         assertTrue(blobStoreName + ": expected supported block size is " + expected + ", but measured: " + test, test >= expected);
     }
 

